<!DOCTYPE html>
<html lang="en-us">
<head>
<meta charset="utf-8">
<meta name="generator" content="Hugo 0.18.1" />
<meta name="viewport" content="width=device-width, initial-scale=1">
<link href="//fonts.googleapis.com/css?family=Roboto:400,700" rel="stylesheet" type="text/css">
<link rel="stylesheet" href="/css/main.css">
<link rel="alternate" href="/index.xml" type="application/rss+xml" title="XIZERO">
<link href="https://cdn.bootcss.com/highlight.js/9.7.0/styles/tomorrow.min.css" rel="stylesheet">
<script src="https://cdn.bootcss.com/highlight.js/9.8.0/highlight.min.js"></script>
<script type="text/javascript" src="http://code.jquery.com/jquery-latest.min.js"></script>
<script src="http://xizero.com/js/top.js"></script>
<script>hljs.initHighlightingOnLoad();</script>

<title>Java 异常 - XIZERO</title>
</head>
<body>

<div id="gotop" title="🚀 go top"></div>

<div class="top">

	<div class="gos">
		<span class="go"><a href="/">Home</a></span>
		<span class="go"><a href="/tags">Tags</a></span>
		<span class="go"><a href="/about">About</a></span>
	</div>
	
	
		<div class="tit">
			Java 异常
		</div>

	

</div>




<div class="con">
	

<h2 id="定义">定义</h2>

<p>编译错误和运行时错误</p>

<h2 id="捕获异常">捕获异常</h2>

<pre><code class="language-java">try {
//会产生异常的代码
} catch (Exception e){
//处理异常的代码
}
finally {
//不管是否产生异常都会执行的代码
}
</code></pre>

<h2 id="捕获多个异常">捕获多个异常</h2>

<p>1.7 支持单一的catch语句中捕获多个异常，每个异常之间使用管道符号 <code>|</code>隔开</p>

<pre><code class="language-java">catch (excepetion-1 | exception-2 ... e) {
    // 相同的处理代码
}
catch (excepetion-n e) {
    //其他的处理代码
}
</code></pre>

<h2 id="try-with-resource">try-with-resource</h2>

<p>1.7 之前的关闭资源时，会使用 finally ，但是关闭时产生异常会嵌套很多层。
1.7 添加 try-with-resource 语句，但是只用实现了 java.lang.AutoCloseable 的类才能关掉。</p>

<pre><code class="language-java">//try ( resource ) { }
try (Statement stmt = con.createStatement()) { }
</code></pre>

<h2 id="java-lang-exception">java.lang.Exception</h2>

<p>Exception 覆盖了 toString () , 返回对异常的说明。
printStackTrace 方法 对 Exception 追踪，打印出对异常的描述。</p>

<p>catch 语句的顺序很重要，JVM 根据其顺序来匹配异常，Exception 的 catch 语句必须放在最后。</p>

<h3 id="throw-和-throws">throw 和 throws</h3>

<p>throw : 用于抛出一个异常对象。
throws：用于方法名之后，表示该方法可能抛出异常的类型。</p>

<h3 id="自定义异常">自定义异常</h3>

<p>通过继承异常类，可以自定义异常。通过throw来抛出自定义异常。</p>

</div>


<div class="boo">
	<span></span>
</div>
<div class="footer">
	@XIZERO Powered by <a href="http://hugo.spf13.com/">Hugo</a> and <a href="#">bigbig</a>
</div>
</body>
</html>

